track/track.go

```diff
@@ -0,0 +1,138 @@
+package track
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+/*
+Track is the interface for card track data
+
+Track provides utility functions for reading and writing magnetic track data
+http://sagan.gae.ucm.es/~padilla/extrawork/tracks.html
+https://en.wikipedia.org/wiki/ISO/IEC_7813
+https://www.q-card.com/about-us/iso-magnetic-stripe-card-standards/page.aspx?id=1457
+
+
+UseCase
+
+message := iso8583.NewMessage(spec)
+message.MTI("0100")
+message.Field(2, "4242424242424242")
+message.Field(3, "123456")
+message.Field(4, "100")
+message.Field(4, "100")
+
+dataBuf, _ := NewTrackSecond().Write(card)
+message.Field(36, string(dataBuf))
+
+*/
+
+type GeneralCard struct {
+	// Available for Track 1, Track 3
+	FormatCode string `json:"format_code"`
+
+	// Available for Track 1, Track 2, Track 3
+	PrimaryAccountNumber string `json:"primary_account_number"`
+
+	// Available for Track 1, Track 2, Track 3
+	CardType string `json:"card_type"`
+
+	// Available for Track 1
+	Name string `json:"name"`
+
+	// Available for Track 1, Track 2
+	ExpirationDate *ExpiryDate `json:"expiration_date"`
+
+	// Available for Track 1, Track 2
+	ServiceCode string `json:"service_code"`
+
+	// Available for Track 1, Track 2, Track 3
+	//  If track 3, the field describe security data + additional data
+	DiscretionaryData string `json:"discretionary_data"`
+}
+
+// Track is network header interface to write/read encoded message legnth
+type Track interface {
```

Comment:
As I suggested in my first comment to the PR, let's implement `field.Field` interface in `Track1` and `Track2`.